### PR TITLE
Decorate class

### PR DESCRIPTION
**Pull request by Esmatullah12**
In this project, I worked on the following tasks
- Implement composition (as an example of the design pattern).
- Run a program using the command line.

### CODE REVIEW

File: capitalize_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+require_relative 'decorator'
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    super.capitalize
+  end
```
Comment (mikethreels) [inline capitalize_decorator.rb]: - [ ] To make this method work you shouldn't call super however you should use the nameable instance variable inherited from the decorator call correct_name and use the capitalize method on that, something like:
```    
    def correct_name
      @nameable.correct_name.capitalize
    end
```

File: trimmer_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+require_relative 'decorator'
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    super[0..9]
+  end
```
Comment (mikethreels) [inline trimmer_decorator.rb]: - [ ] To make this method work you shouldn't call super however you should use the nameable instance variable inherited from the decorator call correct_name and use the spread operator on that, something like:
```
    def correct_name
      @nameable.correct_name[0..9]
    end
```

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi @Esmatullah12, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- good job writing the decorator

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by skyv26 — APPROVED** [review]
## Hi @Esmatullah12, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## STATUS: APPROVED 🟢 



## Highlights
- Implemented the features as per previous code reviewer's request ✅ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

## ⚠️  WARNING ⚠️ 
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Have any doubt ❓ 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @skyv26 in your question so I can receive the notification. You can also follow me on the below platforms

<table>
<tr>
<td>

<a href="https://github.com/skyv26">
   @skyv26
</a>

</td>
<td>

<a href="https://www.linkedin.com/in/skyv2022/">
   linkedin-skyv2022
</a>

</td>
<td>

<a href="https://twitter.com/vrma_aakash">
   @vrma_aakash
</a>

</td>
<td>

<a href="https://www.instagram.com/skybrel/">
   @skybrel
</a>

</td>
<td>

<a href="https://www.frontendmentor.io/profile/skyv26">
   @skybrel
</a>

</td>
</tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**